### DISCUSSION TITLE
how to check if object is proxy or not?

### DISCUSSION THREAD

**Discussion by aelbore** [pmndrs/valtio#473] [Q&A] [answered]
.

**Comment by dai-shi** [comment, marked as answer]
In general, I would suggest to avoid the need of checking if an object is a proxy at runtime. You can't call `useSnapshot` conditionally anyway.
If you need dynamic coding style, you want to have such a property on your end.
```js
import { proxy } from 'valtio'

const IS_PROXY = Symbol()

const state = proxy({
  // ...
  [IS_PROXY]: true,
})

const isProxy = (obj) => obj[IS_PROXY] === true
```

---
For hacks and probably for 3rd-party libraries, this is possible:
```js
import { getVersion } from 'valtio'

const isValtioProxy = (obj) => typeof getVersion(obj) === 'number'
```

**Reply by aelbore** [reply]
@dai-shi thank you appreciate it
